services/movie.py

```diff
@@ -0,0 +1,41 @@
+from typing import Optional
+
+from django.db.models import QuerySet
+
+from db.models import Movie
+
+
+def get_movies(
+        genres_ids: Optional[list[int]] = None,
+        actors_ids: Optional[list[int]] = None
+) -> QuerySet:
+    movies = Movie.objects.all()
+
+    if genres_ids:
+        movies = movies.filter(genres__id__in=genres_ids)
+    if actors_ids:
+        movies = movies.filter(actors__id__in=actors_ids)
+
+    return movies.distinct()
+
+
+def get_movie_by_id(movie_id: int) -> Movie:
+    return Movie.objects.get(pk=movie_id)
```

Comment:
Consider using `Movie.objects.get(pk=movie_id)` within a try-except block to handle the `Movie.DoesNotExist` exception. This will prevent the application from crashing if a movie with the given ID is not found.

Reply:
Just use her get_or_404 ;)